daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml

```diff
@@ -374,16 +509,111 @@
 
   </tdml:parserTestCase>
 
-  <tdml:parserTestCase name="varInstance" root="a"
-    model="v">
+  <tdml:parserTestCase name="varInstance_01" root="nvi1"
+    model="v"
+    description="Simple demonstration of newVariableInstance">
+
+    <tdml:document />
+
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <nvi1 xmlns="http://example.com">
+          <a xsi:type="xsd:int">7</a>
+        </nvi1>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="varInstance_02" root="nvi2"
+    model="v"
+    description="Demonstrates that newVariableInstance correctly goes into and out of scope">
+
+    <tdml:document />
+
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <nvi2 xmlns="http://example.com">
+          <a xsi:type="xsd:int">7</a>
+          <nest1>
+            <b xsi:type="xsd:int">8</b>
+            <nest2>
+              <c xsi:type="xsd:int">9</c>
+            </nest2>
+            <d xsi:type="xsd:int">8</d>
+          </nest1>
+          <e xsi:type="xsd:int">7</e>
+        </nvi2>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="varInstance_03" root="nvi3"
+    model="v"
+    description="Demonstrates using multiple newVariableInstance statements within a sequence">
+
+    <tdml:document />
+
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <nvi3 xmlns="http://example.com">
+          <a xsi:type="xsd:int">7</a>
+          <b xsi:type="xsd:int">17</b>
+          <c xsi:type="xsd:int">27</c>
+        </nvi3>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="varInstance_04" root="nvi4"
+    model="v"
+    description="SDE due to default value being set both as attribute and element value">
+
+    <tdml:document />
+
+    <tdml:errors>
+      <tdml:error>Schema Definition Error</tdml:error>
+      <tdml:error>value of variable was supplied both as attribute and element value</tdml:error>
+    </tdml:errors>
+
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="varInstance_05" root="nvi5"
+    model="v"
+    description="SDE due to multiple newVariableInstance calls within the same scope to the same variable">
+
+    <tdml:document />
+
+    <tdml:errors>
+      <tdml:error>Schema Definition Error</tdml:error>
+      <tdml:error>newVariableInstances must all be distinct within the same scope</tdml:error>
+    </tdml:errors>
+
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase name="varInstance_06" root="nvi6"
+    model="v"
+    description="SDE due to newVariableInstance being declared on an element instead of group reference, sequence or choice">
+
+    <tdml:document />
+
+    <tdml:errors>
+      <tdml:error>Schema Definition Error</tdml:error>
+      <tdml:error>newVariableInstance may only be used on group reference, sequence or choice</tdml:error>
+    </tdml:errors>
+
+  </tdml:parserTestCase>
+
```

Comment:
I don't see tests which verifies that unwinding side-effects on newVariableInstances works properly in the presence of backtracking. 

I suggest a test with two defined variables.

You enter a scope that does NVI on each of them.

You enter another scope that does another NVI on each of them.

Then you enter a choice which sets one and reads the default of the other, then fails, backtracks, and chooses the
next branch, which does the same thing.
The final branch can succeed.

then you unwind to the enclosing scope (different variable instances), and you shoudl be alble to set and read-default-value of those instances, which weren't affected by the backtracking on the inner instances.

Then when you finally unwind all the scopes, the global instances of the variables should be able to set one and read-default from the other, as the global ones should definitely not have been affected.